### PR TITLE
moving BuildTestError and BuildTestCommand methods to utils 

### PR DESCRIPTION
**Pull request by shahzebsiddiqui**
This PR will do the following
 - Move *BuildTestCommand* and *BuildTestError* to ``buildtest/utils/command.py``
 - Fix import statements as necessary 
 - Add regtest for *BuildTestCommand* now. We know this method will be changed but atleast we should test it

Signed-off-by: Shahzeb Siddiqui <shahzebmsiddiqui@gmail.com>

### CODE REVIEW

File: buildtest/utils/file.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 import logging
 
 from buildtest.defaults import logID
-from buildtest.log import BuildTestError
+from buildtest.utils.command import BuildTestError
```
Comment (vsoch) [inline buildtest/utils/file.py]: hmm, should this be under `buildtest.utils.exceptions`?  Or actually, where I usually expect custom exceptions for a library would be is under `buildtest.exceptions` directly. Do you consider this a command?

**Comment by shahzebsiddiqui** [comment]
@vsoch  I know we discussed this, hope this change makes sense.

File: buildtest/utils/file.py (hunk 1 above)
Comment (shahzebsiddiqui) [reply buildtest/utils/file.py]: Sure i can move this over to ``exceptions.py`` give me a second to refactor the code

File: buildtest/utils/command.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        """
+        if args:
+            msg = msg % args
+        self.msg = msg
+
+    def __str__(self):
+        return repr(self.msg)
+
+
+class BuildTestCommand:
```
Comment (vsoch) [inline buildtest/utils/command.py:4]: Love this organization! It's really intuitive to find the `BuildTestCommand` under utils.commands too.

File: buildtest/utils/file.py (hunk 1 above)
Comment (vsoch) [reply buildtest/utils/file.py]: Awesome! And I think it's most intuitive to find at the root (right under buildtest) if that works for you too. I can't tell you how many times I've been writing something to catch an error, and interactively I look for some `<module>.exceptions` to import from.

File: buildtest/buildsystem/singlesource.py [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
 from buildtest.config import config_opts
 from buildtest.defaults import logID
-from buildtest.utils.command import BuildTestError
+from buildtest.utils.exceptions import BuildTestError
```
Comment (vsoch) [inline buildtest/buildsystem/singlesource.py:14]: Can we instead do buildtest.exceptions? That is where I'd expect it.

Comment (shahzebsiddiqui) [reply buildtest/buildsystem/singlesource.py:14]: Sure, let me fix it.

**Review by vsoch — APPROVED** [review]
LGTM, good for merge when all tests pass (green)

**Pull request merged**